invoice/epc_qr.py

```diff
@@ -0,0 +1,123 @@
+"""EPC QR code generation utilities."""
+
+from decimal import Decimal
+from typing import Optional
+
+from schwifty import BIC, IBAN
+
+_VERSIONS = ("001", "002")
+_ENCODINGS = {
+    "1": {"utf8", "utf-8"},
+    "2": {"latin", "latin1", "latin-1", "iso8859-1", "iso-8859-1"},
+    "3": {"latin2", "iso8859-2", "iso-8859-2"},
+    "4": {"latin4", "iso8859-4", "iso-8859-4"},
+    "5": {"cyrillic", "iso8859-5", "iso-8859-5"},
+    "6": {"greek", "iso8859-7", "iso-8859-7"},
+    "7": {"latin6", "iso8859-10", "iso-8859-10"},
+    "8": {"latin9", "iso8859-15", "iso-8859-15"},
+}
+
+
+def gen_epc_qr_data(beneficiary_name: str, beneficiary_iban: str, beneficiary_bic: Optional[str] = None,
```

Comment:
I want to remove spaces if someone passes in an IBAN string with spaces.
When passing in an IBAN object this still works.
This method should be totally independent of our models.

Reply:
But why? It will not used in a different context. For a standalone module I'll agree but this is a end customer app.

Reply:
It is stand-alone, its not part of the invoice or bank account model, even though it could be.
Furthermore the spec requires that IBANs should not contain spaces.

I need an IBAN object inside the function either way to get the BIC via schwifty - which is an implementation detail and not part of the API.
This internal detail should not bleed through.

Even if I add type annotations for schwifty.IBAN this requires explicit `isinstance(IBAN)` checks to not break - but those are not really pythonic.

If I say only strings allowed, I need to do a clean_text and replace spaces.

So both avenues are more complicated code-wise. Which led me to the current code - unconditionally converting to IBAN again, which works all the time and is simple code.

Reply:
Great explanation. I get it now. Let's do it as you suggested.